scitran/data/meeg/meeg.py

```diff
@@ -0,0 +1,190 @@
+
+
+class MEEGError(data.DataError):
+    pass
+
+
+class MEEGReader(data.Reader):
+
+    """
+    Parameters
+    ----------
+    path : str
+        Path to input file.
+    load_data : boolean
+        Indicate if a reader should attempt to immediately load all data.
+        Default False.
+    """
+    project_properties = project_properties
+    session_properties = session_properties
+    acquisition_properties = acquisition_properties
+
+    domain = u'meeg'
+    filetype = u'meeg'
+    state = ['orig']
+
+    def __init__(self, path, load_data=False, timezone=None):
+        super(MEEGReader, self).__init__(path, load_data, timezone)
+        self._temp_dir = tempfile.mkdtemp()
+        os.mkdir(op.join(self._temp_dir, 'reap'))
+        try:
+            with zipfile.ZipFile(self.filepath, 'r') as zip_file:
+                hdr = json.loads(zip_file.comment,
+                                 object_hook=util.datetime_decoder)['header']
+                self.project_name = hdr['project']
+                zip_fnames = [op.join('reap', op.basename(fname))
+                              for fname in zip_file.namelist()]
+                fnames = []
+                for fname in zip_fnames:
+                    if fname.endswith('.fif'):
+                        fnames.append(zip_file.extract(fname, self._temp_dir))
+                # load raw files
+                with warnings.catch_warnings(record=True):
+                    self._raws = [read_raw_fif(fname, allow_maxshield=True,
+                                               preload=load_data)
+                                  for fname in fnames]
+        except Exception as e:
+            raise MEEGError(e)
+
+    def __del__(self):
+        shutil.rmtree(self._temp_dir)
+
+    def load_data(self):
+        super(MEEGReader, self).load_data()
+        for raw in self._raws:
+            raw.preload_data()
+
+    @property
+    def nims_project(self):
+        return self.project_name
+
+    @property
+    def nims_group_id(self):
+        # XXX
+        return 'group_id'
+
+    @property
+    def nims_session_id(self):
+        # XXX
+        return 'session_id'
+
+    @property
+    def nims_session_label(self):
+        # XXX
+        return 'session_label'
+
+    @property
+    def nims_session_subject(self):
+        # XXX
+        return 'subj_code'
+
+    @property
+    def nims_acquisition_label(self):
+        # XXX
+        return 'series_no'
+
+    @property
+    def nims_acquisition_description(self):
+        # XXX get from info
+        return 'foobar'
+
+    @property
+    def nims_file_name(self):
+        # XXX what are the constraints on this?
+        return 'foo'
+
+    @property
+    def nims_file_kinds(self):
+        # XXX
+        return ['FIF']
+
+    @property
+    def nims_acquisition_id(self):
+        # XXX
+        return 'nims_acquisition_id'
+
+    # XXX why are these not implemented at the data.Reader level?
```

Comment:
@gsfr any reason not to have the following as part of the super `reader.Reader` class? It would make the code more DRY to have these as defaults in that class. But if you want `Reader` to be an ABC I guess it makes sense.